test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,493 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .orderBy<string>('order as long', 'DESC').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.reverse().map(function(obj, index) {
+        index += 1;
+        expect(obj.order).to.be.equal(index);
+      });
+
     });
 
-    it('should query by endsWith', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereEndsWith('name', '7').get();
-      
-      expect(results[0].name).to.equals('test107');
+    it('should query with whereGreaterThanOrEqual', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereGreaterThanOrEqual<number>('order',6, 'name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+        results.map(function(obj) {
+
+          expect(obj).to.include({'name': 'withNesting'});
+
+        });
+
     });
 
-    it('should fail query with non-existing index', async () => expect(
-        store.openSession().query({
-          indexName: 's'
-        }).where({'Tag': 'Products'}).get()
-      ).to.be.rejected
+    it('should query with whereLessThanOrEqual', async () => {
+      const results: Universal[] = await  store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereLessThanOrEqual<number>('order',6, 'name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function(obj) {
+
+        expect(obj).to.include({'name': 'withNesting'});
+
+      });
+
+    });
+
+    it('should query with nested objects', async () => {
+      const results: Universal[] = await store.openSession().query<Universal>({
+        indexName: 'Universals',
+        nestedObjectTypes: {product: Product}
+      }).whereEquals<string>('name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function (obj) {
+        expect(obj.name).to.equal('withNesting');
+        expect(obj.product.name).to.equal('testing_order');
+      });
+
+    });
+
+    it('should make query with fetch terms', async () => {
+      const results: Universal[] = await store.openSession().query<Universal>({
+        WaitForNonStaleResults: true,
+        indexName: 'Universals'
+      }).selectFields('id').get();
+
+      expect(results).to.have.lengthOf(10);
+      expect(_.every(results, (result: Product) => result.hasOwnProperty('id'))).to.be.true;
+
+    });
+
+    it('should query with whereArray', async () => expect(
+      store.openSession().query<Universal>({
+        indexName: 'Universals'
+      }).where({
+        name: [4, 6, 90]
+      }).get()
+      ).to.be.fulfilled
```

Comment:
It's just checking whether the query succeeded. Its results should be verified with proper assertions.